src/polytope_blmos.jl

```diff
@@ -154,20 +181,41 @@ function bounded_compute_extreme_point(
     indices = collect(1:length(d))
     perm = sortperm(d)
 
-    # The lower bounds always have to be met. 
+    # Step 1: satisfy integer lower bounds
     v[int_vars] = lb
 
+    # Step 2: distribute remaining N
     for i in indices[perm]
+        rem = sblmo.N - sum(v)
+        if rem ≤ 1e-10
+            break
+        end
+
         if i in int_vars
-            idx = findfirst(x -> x == i, int_vars)
-            v[i] += min(ub[idx] - lb[idx], sblmo.N - sum(v))
+            idx = findfirst(==(i), int_vars)
+            add_int = min(ub[idx] - v[i], floor(rem))  # make sure int
+            v[i] += add_int
         else
-            v[i] += sblmo.N - sum(v)
+            # continous variable can be the rem
+            v[i] += rem
+        end
+    end
+
+    # make sure sum is N
+    rem = sblmo.N - sum(v)
+    if abs(rem) > 1e-8
+        cont_vars = setdiff(indices, int_vars)
+        if !isempty(cont_vars)
+            v[cont_vars[1]] += rem
+        else
```

Comment:
This should never be the case! Also, just assigning the remainder to the first continuous variable might not lead to the truly optimal solution.